diff --git a/node_modules/react-native-video/android/.classpath b/node_modules/react-native-video/android/.classpath
new file mode 100644
index 0000000..bbe97e5
--- /dev/null
+++ b/node_modules/react-native-video/android/.classpath
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-17/"/>
+	<classpathentry kind="con" path="org.eclipse.buildship.core.gradleclasspathcontainer"/>
+	<classpathentry kind="output" path="bin/default"/>
+</classpath>
diff --git a/node_modules/react-native-video/android/.gradle/8.9/checksums/checksums.lock b/node_modules/react-native-video/android/.gradle/8.9/checksums/checksums.lock
new file mode 100644
index 0000000..6e94778
Binary files /dev/null and b/node_modules/react-native-video/android/.gradle/8.9/checksums/checksums.lock differ
diff --git a/node_modules/react-native-video/android/.gradle/8.9/dependencies-accessors/gc.properties b/node_modules/react-native-video/android/.gradle/8.9/dependencies-accessors/gc.properties
new file mode 100644
index 0000000..e69de29
diff --git a/node_modules/react-native-video/android/.gradle/8.9/fileChanges/last-build.bin b/node_modules/react-native-video/android/.gradle/8.9/fileChanges/last-build.bin
new file mode 100644
index 0000000..f76dd23
Binary files /dev/null and b/node_modules/react-native-video/android/.gradle/8.9/fileChanges/last-build.bin differ
diff --git a/node_modules/react-native-video/android/.gradle/8.9/fileHashes/fileHashes.lock b/node_modules/react-native-video/android/.gradle/8.9/fileHashes/fileHashes.lock
new file mode 100644
index 0000000..e6d42f8
Binary files /dev/null and b/node_modules/react-native-video/android/.gradle/8.9/fileHashes/fileHashes.lock differ
diff --git a/node_modules/react-native-video/android/.gradle/8.9/gc.properties b/node_modules/react-native-video/android/.gradle/8.9/gc.properties
new file mode 100644
index 0000000..e69de29
diff --git a/node_modules/react-native-video/android/.gradle/buildOutputCleanup/buildOutputCleanup.lock b/node_modules/react-native-video/android/.gradle/buildOutputCleanup/buildOutputCleanup.lock
new file mode 100644
index 0000000..5578565
Binary files /dev/null and b/node_modules/react-native-video/android/.gradle/buildOutputCleanup/buildOutputCleanup.lock differ
diff --git a/node_modules/react-native-video/android/.gradle/buildOutputCleanup/cache.properties b/node_modules/react-native-video/android/.gradle/buildOutputCleanup/cache.properties
new file mode 100644
index 0000000..15c8d9d
--- /dev/null
+++ b/node_modules/react-native-video/android/.gradle/buildOutputCleanup/cache.properties
@@ -0,0 +1,2 @@
+#Sun Aug 09 18:01:45 IST 2026
+gradle.version=8.9
diff --git a/node_modules/react-native-video/android/.gradle/vcs-1/gc.properties b/node_modules/react-native-video/android/.gradle/vcs-1/gc.properties
new file mode 100644
index 0000000..e69de29
diff --git a/node_modules/react-native-video/android/.project b/node_modules/react-native-video/android/.project
new file mode 100644
index 0000000..b5e2d8f
--- /dev/null
+++ b/node_modules/react-native-video/android/.project
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>react-native-video</name>
+	<comment>Project react-native-video created by Buildship.</comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
+	</natures>
+</projectDescription>
diff --git a/node_modules/react-native-video/android/.settings/org.eclipse.buildship.core.prefs b/node_modules/react-native-video/android/.settings/org.eclipse.buildship.core.prefs
new file mode 100644
index 0000000..1675490
--- /dev/null
+++ b/node_modules/react-native-video/android/.settings/org.eclipse.buildship.core.prefs
@@ -0,0 +1,2 @@
+connection.project.dir=../../../android
+eclipse.preferences.version=1
diff --git a/node_modules/react-native-video/lib/specs/VideoNativeComponent.js b/node_modules/react-native-video/lib/specs/VideoNativeComponent.js
index 0db7309..d71d056 100644
--- a/node_modules/react-native-video/lib/specs/VideoNativeComponent.js
+++ b/node_modules/react-native-video/lib/specs/VideoNativeComponent.js
@@ -1,5 +1,16 @@
 "use strict";
 Object.defineProperty(exports, "__esModule", { value: true });
 const react_native_1 = require("react-native");
-exports.default = (0, react_native_1.requireNativeComponent)('RCTVideo');
+
+const isNativeAvailable = react_native_1.UIManager.getViewManagerConfig 
+    ? react_native_1.UIManager.getViewManagerConfig('RCTVideo') != null 
+    : react_native_1.UIManager.RCTVideo != null;
+
+class FallbackVideo extends react_native_1.Component {
+  render() { 
+      return react_native_1.createElement(react_native_1.View, this.props); 
+  }
+}
+
+exports.default = isNativeAvailable ? (0, react_native_1.requireNativeComponent)('RCTVideo') : FallbackVideo;
 //# sourceMappingURL=VideoNativeComponent.js.map
diff --git a/node_modules/react-native-video/src/specs/VideoNativeComponent.ts b/node_modules/react-native-video/src/specs/VideoNativeComponent.ts
index 3dcbe01..1cb8f3b 100644
--- a/node_modules/react-native-video/src/specs/VideoNativeComponent.ts
+++ b/node_modules/react-native-video/src/specs/VideoNativeComponent.ts
@@ -1,6 +1,7 @@
 /* eslint-disable @typescript-eslint/ban-types */
 import type {HostComponent, ViewProps} from 'react-native';
-import {requireNativeComponent} from 'react-native';
+import {requireNativeComponent, UIManager, View} from 'react-native';
+import React, { Component } from 'react';
 import type {
   DirectEventHandler,
   Double,
@@ -414,6 +415,28 @@ export interface VideoNativeProps extends ViewProps {
 
 type NativeVideoComponentType = HostComponent<VideoNativeProps>;
 
-export default requireNativeComponent<VideoNativeProps>(
+let isNativeAvailable = false;
+try {
+  if (UIManager.getViewManagerConfig) {
+    isNativeAvailable = UIManager.getViewManagerConfig('RCTVideo') != null;
+  } else {
+    isNativeAvailable = (UIManager as any).RCTVideo != null;
+  }
+} catch (e) {
+  isNativeAvailable = false;
+}
+
+class FallbackVideo extends Component<VideoNativeProps> {
+  seek() {}
+  resume() {}
+  pause() {}
+  setFullScreen() {}
+  save() {}
+  render() { return React.createElement(View, this.props); }
+}
+
+export default (isNativeAvailable ? requireNativeComponent<VideoNativeProps>(
   'RCTVideo',
-) as NativeVideoComponentType;
+) : FallbackVideo) as unknown as NativeVideoComponentType;
+
+
